plugins/signalmonitor/signalmonitor.cpp

```diff
@@ -40,14 +40,32 @@
 #include <QTimer>
 
 using namespace GammaRay;
+typedef ServerProxyModel<QSortFilterProxyModel> ServerFilterProxyModel;
+
+class SignalMonitorProxyModel : public ServerFilterProxyModel
+{
+public:
+    using ServerFilterProxyModel::ServerFilterProxyModel;
+
+    void sort(int column, Qt::SortOrder order = Qt::AscendingOrder) override
+    {
+        if (static_cast<SignalHistoryModel::ColumnId>(column) == SignalHistoryModel::MonitoredColumn) {
+            setSortRole(Qt::CheckStateRole);
```

Comment:
The sort implementation is done in the dptr, setSortRole() only call d->sort(), not QAIM::sort().